pkg/autodiscovery/configresolver/configresolver.go

```diff
@@ -48,6 +48,10 @@ func Resolve(tpl integration.Config, svc listeners.Service) (integration.Config,
 	copy(resolvedConfig.InitConfig, tpl.InitConfig)
 	copy(resolvedConfig.Instances, tpl.Instances)
 
+	if resolvedConfig.IsCheckConfig() && !svc.IsReady() {
+		return resolvedConfig, errors.New("unable to resolve, service not ready")
```

Comment:
nits, should you add a service meta, like a name/id ?

Reply:
would be redundant with this though 🤔 
https://github.com/DataDog/datadog-agent/blob/2e76d3c73e9a99f3fcd1560a851efa4406212ebe/pkg/autodiscovery/autoconfig.go#L527

Reply:
👍